### ISSUE TITLE
State-of-the-art loss functions

### ISSUE THREAD

**Issue by mratsim** [mratsim/Arraymancer#277]
- [ ] CTC loss #269
- [ ] LovaszSoftmax loss https://github.com/bermanmaxim/LovaszSoftmax and in TensorFlow
  ```Python
  def lovasz_hinge(logits, labels, per_image=True, ignore=None):
      """
      Binary Lovasz hinge loss
        logits: [B, H, W] Variable, logits at each pixel (between -\infty and +\infty)
        labels: [B, H, W] Tensor, binary ground truth masks (0 or 1)
        per_image: compute the loss per image instead of per batch
        ignore: void class id
      """
      if per_image:
          def treat_image(log_lab):
              log, lab = log_lab
              log, lab = tf.expand_dims(log, 0), tf.expand_dims(lab, 0)
              log, lab = flatten_binary_scores(log, lab, ignore)
              return lovasz_hinge_flat(log, lab)
        
          losses = tf.map_fn(treat_image, (logits, labels), dtype=tf.float32)
          #losses.set_shape((None,))
          loss = tf.reduce_mean(losses) 
      else:
          loss = lovasz_hinge_flat(*flatten_binary_scores(logits, labels, ignore))
      return loss
    ```
    See use in [TGS Salt identification Challenge](https://www.kaggle.com/c/tgs-salt-identification-challenge) for the Oil & Gas industry.
- [ ] Fbeta loss functions (for multilabel classifications)
